### ISSUE TITLE
Dart/Flutter SDK compatibility - streams

### ISSUE THREAD

**Issue by vaind** [objectbox/objectbox-dart#197]
## TLDR

If you don't use query stream (`Query.findStream`) or sync event streams (`SyncClient.*Events`), you can use the latest version of ObjectBox, with any recent version of Flutter (v1.20, v1.22, v2.0) or Dart (v2.10, v2.12)

Otherwise, you need to make a choice:

* either use the latest ObjectBox v0.12 with the previous version of Flutter (v1.20 or v1.22) or Dart (v2.10) - this gives you query streams and sync event streams.
* or use the previous ObjectBox v0.11.1 with any recent version of Flutter (v1.20, v1.22, v2.0) or Dart (v2.10, v2.12) - this gives you query streams but no sync event streams and also not the latest features.

We're working on a new release for Flutter v2.0 and Dart v2.12 which will have to break compatibility with older versions (due Dart SDK changes in FFI).

## More details

Implementing observers/query streams and sync event streams requires asynchronous callbacks from the native library to the current dart isolate. [Currently, Dart provides limited support for this](https://github.com/dart-lang/sdk/issues/37022) - the native library communicates directly with the Dart VM. The drawback of this temporary solution (until Dart SDK provides a final one) is that whenever Dart SDK changes the VM API in a new release, the library (objectbox) has to do that as well, switching to the new API. 

## Version support

* old objectbox v0.11 - didn't yet use native async callbacks, and didn't use the Dart VM API, is therefore compatible with:
  - Flutter v1.17+
  - Dart v2.8+
  - and should also work with some prerelease versions (note: your mileage may vary, not tested with pre-releases)

* current objectbox v0.12 introduces full support to native async callbacks and is compatible with:
  - Flutter v1.20
  - Flutter v1.22
  - Dart v2.9
  - Dart v2.10

* future objectbox v0.13 (or maybe higher, depends on the next Dart/Flutter SDK release) 
  - will be compatible with the new Dart/Flutter SDK release but will drop compatibility with the previous one, due to Dart VM API changes

**Comment by vaind** [comment]
with the released v0.13, the new SDK interoperability API is supported in Flutter v2.0 and Dart 2.12.
